studio/components/interfaces/Auth/SmtpForm/SmtpForm.tsx

```diff
@@ -110,26 +115,31 @@ const SmtpForm = () => {
     }),
   })
 
-  const onSubmit = async (values: any, { setSubmitting, resetForm }: any) => {
+  const onSubmit = (values: any, { setSubmitting, resetForm }: any) => {
     const payload = enableSmtp ? values : defaultDisabledSmtpFormValues
 
     // Format payload: Remove redundant value + convert port to string
     delete payload.ENABLE_SMTP
     payload.SMTP_PORT = payload.SMTP_PORT ? payload.SMTP_PORT.toString() : payload.SMTP_PORT
 
     setSubmitting(true)
-    const { error } = await authConfig.update(payload)
-
-    if (!error) {
-      setHidden(true)
-      const updatedFormValues = generateFormValues(payload)
-      resetForm({ values: updatedFormValues, initialValues: updatedFormValues })
-      ui.setNotification({ category: 'success', message: 'Successfully updated settings' })
-    } else {
-      ui.setNotification({ category: 'error', message: 'Failed to update settings', error })
-    }
-
-    setSubmitting(false)
+    updateAuthConfig(
+      { projectRef: projectRef!, config: payload },
+      {
+        onError: (error) => {
```

Comment:
likewise as per https://github.com/supabase/supabase/pull/16227/files#r1289813556
shift all these to the mutation declaration

use isLoading from the mutation rather than isSubmitting from the Form

Reply:
Can't, due to `resetForm`.  I've removed `isSubmitting`.